File: src/main/scala/info/crlog/higgs/protocols/boson/BosonEncoder.scala
package info.crlog.higgs.protocols.boson

import io.netty.handler.codec.MessageToByteEncoder
import io.netty.channel.ChannelHandlerContext
import io.netty.buffer.ByteBuf

/**
 * @author Courtney Robinson <courtney@crlog.info>
 */
class BosonEncoder extends MessageToByteEncoder[Array[Byte]] {
  def encode(ctx: ChannelHandlerContext, msg: Array[Byte], out: ByteBuf) {
    //protocol version, message size and message are already in the byte array so just write it out
    out.writeBytes(msg)
  }
}

File: higgs-boson/src/main/java/com/fillta/higgs/boson/serialization/BosonEncoder.java
package com.fillta.higgs.boson.serialization;

import io.netty.buffer.ByteBuf;
import io.netty.channel.ChannelHandlerContext;
import io.netty.handler.codec.MessageToByteEncoder;

/**
 * @author Courtney Robinson <courtney@crlog.info>
 */
public class BosonEncoder extends MessageToByteEncoder<ByteBuf> {
	@Override
	public void encode(ChannelHandlerContext ctx, ByteBuf msg, ByteBuf out) throws Exception {
		out.writeBytes(msg);
		ctx.flush();
	}
}
